gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStoreUtils.js

```diff
@@ -0,0 +1,408 @@
+/*!
+GPII CouchDB OAuth 2 Data Store
+
+Copyright 2016 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    $ = fluid.registerNamespace("jQuery"),
+    uuid = require("node-uuid");
+
+require("../../gpii-oauth2-utilities/src/OAuth2Utilities.js");
+
+fluid.registerNamespace("gpii.oauth2.dbDataStore");
+
+// All doc types used for saving different records into CouchDB/PouchDB
+gpii.oauth2.dbDataStore.docTypes = {
```

Comment:
There's a lot of stuff in this file and it's not very clear what it does. It could benefit from a quite large "strategy" comment at the head, listing all the different record types and their expected members, as well as a general explanation for the role of the file contents. Each utility should have a comment explaining its expected arguments and return and (most importantly) whether any of its arguments are modified.